webpack.config.js

```diff
@@ -0,0 +1,217 @@
+
+const path = require('path');
+const webpack = require('webpack');
+const autoprefixer = require('autoprefixer');
+const FileManagerPlugin = require('filemanager-webpack-plugin');
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const OptimizeCSSAssetsPlugin = require("optimize-css-assets-webpack-plugin");
+const WebpackAssetsManifest = require("webpack-assets-manifest");
+const ExtraWatchWebpackPlugin = require("extra-watch-webpack-plugin");
+const FixStyleOnlyEntriesPlugin = require("webpack-fix-style-only-entries");
+const glob = require('glob')
+
+// /////////////////////////////////////////////////////////////////////////////
+// Paths ///////////////////////////////////////////////////////////////////////
+// /////////////////////////////////////////////////////////////////////////////
+
+const npmPackage = 'node_modules';
+const srcDir = path.resolve(__dirname, "lib");
+const distDir = path.resolve(__dirname, "dist");
+const srcSass = path.resolve(__dirname, srcDir, "scss");
+const distSass = path.resolve(__dirname, distDir, "css");
+const srcJS = path.resolve(__dirname, srcDir, "js");
+const distJS = path.resolve(__dirname, distDir, "js");
+
+// /////////////////////////////////////////////////////////////////////////////
+// Functions ///////////////////////////////////////////////////////////////////
+// /////////////////////////////////////////////////////////////////////////////
+
+// /////////////////////////////////////////////////////////////////////////////
+// Config //////////////////////////////////////////////////////////////////////
+// /////////////////////////////////////////////////////////////////////////////
+
+
+const entryPoints = glob.sync('./lib/scss/**/*.scss').reduce((acc, filePath) => {
+  const filePathParts = filePath.replace('./lib/scss/', '').split('/');
+  let fileName = filePathParts.pop();
+  if (fileName.indexOf('_') === 0) {
+    return acc;
+  }
+  if (fileName === 'index.scss') {
+    fileName = filePathParts.pop();
+  }
+  const entry = filePathParts.length >= 1 ? filePathParts.join('/') + '/' + fileName : fileName;
+  acc[entry.replace('.scss', '')] = filePath
+  return acc
+}, {});
+
+// Start configuring webpack.
+var webpackConfig = {
+  // What am i?
+  name: 'stanford_publication',
+  // Allows for map files.
+  devtool: 'source-map',
+  // What build?
+  entry: entryPoints,
+
+  // Where put build?
+  output: {
+    filename: "[name].js",
+    path: path.resolve(__dirname, distJS),
+  },
+  // Relative output paths for css assets.
+  resolve: {
+    alias: {
+      'decanter-assets': path.resolve(npmPackage, 'decanter/core/src/img'),
+      'decanter-src': path.resolve(npmPackage, 'decanter/core/src'),
+      '@fortawesome': path.resolve(npmPackage, '@fortawesome'),
+      'fa-fonts': path.resolve(npmPackage, '@fortawesome/fontawesome-free/webfonts')
+    }
```

Comment:
Haven't needed the alias option as much lately as we've been using the `~` practice.

Reply:
I think I grabbed this all from stanford profile styles. We might need updating a few places

Reply:
Jah, not a big deal and won't hurt.

Reply:
I grabbed the latest from the module example for this :+1: